wrappers/binaries_test.go

```diff
@@ -226,14 +358,15 @@ func (s *binariesTestSuite) testAddSnapBinariesAndRemove(c *C, useLegacy bool, d
 		compDir = dirs.LegacyCompletersDir
 	}
 	if disabledCompletion {
-		c.Assert(osutil.FileExists(dirs.CompleteShPath(s.base)), Equals, false)
 		c.Assert(osutil.IsSymlink(filepath.Join(dirs.CompletersDir, "hello-snap.world")), Equals, false)
 		c.Assert(osutil.IsSymlink(filepath.Join(dirs.LegacyCompletersDir, "hello-snap.world")), Equals, false)
-	} else if osutil.FileExists(dirs.CompleteShPath(s.base)) {
+	} else {
+		c.Assert(osutil.FileExists(dirs.CompleteShPath(s.base)), Equals, true)
```

Comment:
I think the tests were confusing originally, and the meaning of `disabledCompletion` was not clear.

There are 3 ways to disable completion:
 * bash-completion is not installed, or has no version
 * the base does not have completion script (like a very old core)
 * the snap does not have completion

I think the intention for `disabledCompletion` was for the first case.
The second case could be detected with `osutil.FileExists(dirs.CompleteShPath(s.base))`.
And for the last case, the snap always had completion.

It looks to me that some tests were testing the cases where base snap was missing support and system was missing completion. But we were missing the cases where snap had support and system did not.
 
You added in those tests: 
```
c.Assert(ioutil.WriteFile(dirs.CompleteShPath(s.base), nil, 0644), IsNil)
```

So now it looks like we have cases where snap has support and system does not. But we lost the case where both did not have support. I think it would be better to have both cases. Make sure `disabledCompletion` means that the system does not have support (we can rename it). And then change this function to:
```
if disabledCompletion || !osutil.FileExists(dirs.CompleteShPath(s.base)) {
   // test the completers are missing
} else {
   // test all the completers are here
}
```

Reply:
Actually. Thinking more about it, you do not need to add the cases where it is disabled by both. It is not going to cover more code.

But I still would like the `disabledCompletion` to be renamed to something like `disabledCompletionOnHost` and use that if condition.

Reply:
It is also possible to move that flag to field in `binariesTestSuite`.

Reply:
totally agree, it would be much readable this way, thanks!